work_with_git.py

```diff
@@ -0,0 +1,40 @@
+import git
+from git import Repo
+from git import Actor
+
+
+author = Actor("An author", "katridi@yandex.ru")
+committer = Actor("A committer", "katridi@yandex.ru")
+
+def clone_repo(url, path):
```

Comment:
Looks like we are using repo as first argument in all methods besides this, so we can create our own proxy class and convert clone to factory `classmethod`. Pseudo-code below:
```python
class Repo:
    @classmethod
    def clone(cls, path):
        .... returning instance of cls
```

Reply:
Thanks, I'll consider it.

Reply:
For now I'll wait for others, but your suggestion seems reasonable enough!

Reply:
Could you explain to me and possibly to others why @classmethod is more preferable than this snipet:
https://gist.github.com/jekoff/3ce51119a76b01345d97c72da99c1c52
Here one initializes self.repo and then do whatever he need
Thanks in advance!

Reply:
It is better to keep `__init__` method as simple as possible. So any actions beyond the object construction itself should be separated. It is my personal opinion for sure. Just compare usage snippets:
```python
repo = Repository('git@github.com:sphericalpm/lektorium.git', '/tmp/lektorium')
```
```python
repo = Repository.clone('git@github.com:sphericalpm/lektorium.git', '/tmp/lektorium')
```
In second example you clearly show the action to be taken. Explicit is better than implicit!

Reply:
Fair enough, thanks!